cmd/juju/user/login.go

```diff
@@ -27,12 +27,11 @@ import (
 )
 
 const loginDoc = `
-When a known domain is specified instead of a username or the
---host flag is specified the user is logged into
-the controller associated with the given domain and the controller is registered
-using the domain name (the -c flag can be used to choose a different controller
-name). If a different controller is already registered with the same name,
-it is an error.
+By default, the juju login command logs the user into a controller
+(either a known public controller or the host name of a public
+controller). In this case, the -c flag can be used to choose a name for
+the new controller. The -u flag causes it to log into a controller as a
+local user instead - the -c flag names a current controller in this case.
```

Comment:
I think this might be clearer for the above paragraph:

> By default, the juju login command logs the user into a public controller. If the controller is already known it can be specified by name. Alternatively, the host name of a public controller can be specified. The -c flag can be used to specify a name for the controller when a host name is provided.
> 
> If the -u flag is provided, the juju login command will attempt to log into a controller as a local user. In this case, the -c flag names the controller to log in to.

This still assumes a fair bit of knowledge (e.g. the "public controller" and "local user" concepts) but I think it's clearer.